javascript/ql/experimental/adaptivethreatmodeling/lib/experimental/adaptivethreatmodeling/EndpointFeatures.qll

```diff
@@ -184,14 +184,26 @@ module NeighborhoodBodies {
    * in the subtree.
    */
   Raw::AstNode getNeighborhoodAstNode(Raw::AstNode node) {
-    if getNumDescendents(node.getParentNode()) > maxNumDescendants()
+    if
+      // `node` will always have a parent as we start at and endpoint
+      node.getParentNode() = getOutermostEnclosingFunction(node) or
```

Comment:
Doesn't this mean the neighborhood can _never_ be the entire enclosing function? We could instead do
```
if
      node = getOutermostEnclosingFunction(node) or
...
```

Reply:
Yep, and this does happen in this example: https://github.com/wilsto/BoardOS/blob/develop/client/app/KPI/KPI.controller.js#L96-L100

Reply:
This emulates what happens in `getTokenBodyFeatureForEntity`, which also returns the function body but not the top-level function AST node (i.e. the function name + parameters I think). 

In the above example, `neighborhoodBody` is `<ABSENT>` but `enclosingFunctionBody` is also really short (and would be identical to `neighborhoodBody`).

Reply:
I have a few questions about this:
1. If this emulates what happens in `getTokenBodyFeatureForEntity` then why does `neighborhoodBody` end up `<ABSENT>` while `enclosingFunctionBody` is short but not absent?
2. Why would we want `neighborhoodBody` to be `<ABSENT>` rather than being short and identical to `enclosingFunctionBody`? There could still be useful signal in the short sequence. Different features are have different paths through the network (different parameters are learned for each), so having identical values in some instances isn't redundant. Also, we're hoping we may be able to replace the full function body with these more localized features eventually.